### PR TITLE
[FrameworkBundle] Allow to disable the PropertyAccessor and annotations

### PR DESCRIPTION
**Pull request by GuilhemN**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | - |
| Fixed tickets | https://github.com/symfony/symfony/issues/13703#issuecomment-178670943 |
| License | MIT |

Includes #17690

This PR allows to disable `framework.property_access` and `framework.annotations` and deprecate automatically enabling them.

### CODE REVIEW

**Comment by Ener-Getick** [comment]
I'm waiting for feedbacks before fixing all deprecation notices in the core.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -523,10 +533,20 @@ private function addSerializerSection(ArrayNodeDefinition $rootNode)
     private function addPropertyAccessSection(ArrayNodeDefinition $rootNode)
     {
         $rootNode
+            ->beforeNormalization()
+                ->always(function ($v) {
+                    if (!isset($v['property_access'])) {
+                        $v['property_access']['enabled'] = true;
+                        @trigger_error('You must explicitly define wheter to enable "framework.property_access" or not. It will be disabled by default in 4.0', E_USER_DEPRECATED);
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: not sure if that's a good idea

Comment (Ener-Getick) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: As discussed https://github.com/symfony/symfony/issues/13703, it is confusing to have some components enabled by default and some others disabled by default.
BTW this deprecation can be easily fixed in the symfony standard edition.

**Comment by fabpot** [comment]
This one should be rebased once #17690 is merged.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php (hunk 1 above)
Comment (Ener-Getick) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: I commented it for now as there are a lot of changes to make to fix all the deprecations in the core then ...

**Comment by Ener-Getick** [comment]
@fabpot I rebased the PR and commented the deprecations notices for now as there are a lot of deprecations to fix in the core...
I'm wondering if this PR should deprecate enabling ``property_access`` and ``annotations`` by default.

**Comment by Ener-Getick** [comment]
I removed the deprecations to have something easier to review as we can still add them later.

**Comment by Ener-Getick** [comment]
I'm closing this for now as the conflicts are too complex to fix.

**Pull request closed without merging**